### ISSUE TITLE
Couldn't parse this $((..)) expression.

### ISSUE THREAD

**Issue by orbea** [koalaman/shellcheck#999]
#### For bugs
- Rule Id: `SC1073`, `SC1009` and `SC1072`
- My shellcheck version: `online`
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

set -eu

test='415670712'
math='/1024*21/10'

echo "$(((${test}+512)$math))" # 852448

```

#### Here's what shellcheck currently says:

```
Line 8:
echo "$(((${test}+512)$math))" # 852448
      ^-- SC1073: Couldn't parse this $((..)) expression.
      ^-- SC1009: The mentioned parser error was in this $((..)) expression.
                      ^-- SC1072: Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:

Nothing.

The example correctly works in any shell I have tested it with and shellcheck should be able to parse it.

**Comment by kurahaupo** [comment]
TL;DR - working as intended.

"Works in shell" and "Shellcheck should parse" are two different things.

The `$math` inside (( )) can only be parsed after `math` has been set at run-time.

Shellcheck is a **_static_ analyzer**, and advertised as such. That means that it does _not_ follow the flow of control that the shell would execute.

One consequence of this is that Shellcheck may not know the values of variables, functions and aliases where they are expanded or called; so it rightly takes a pessimistic view, and warns you.

If you really know what you're doing, disable the warning with
```
# shellcheck disable=SC1073
```

**Comment by orbea** [comment]
If this is working as intended then shellcheck is inherently buggy software... I really hope that is not the goal of this project.

**Comment by orbea** [comment]
According to the documentation.
```
If you're getting a parser error for code you know or think is correct, you should submit a bug with an example!
```
https://github.com/koalaman/shellcheck/wiki/Parser-error

The code is correct in that it works and is portable.

**Comment by koalaman** [comment]
ShellCheck could try to determine the span of the `$(( .. ))` expression first, and parse the contents as an arithmetic expression afterwards. This is basically how shells do it, and how ShellCheck already parses `` `..` ``.

It would still result in a parsing error, but instead of aborting it could emit a warning and continue. The `$((..))` would remain as a black box for the purpose of variable usage and analysis.

You can already do this manually as a workaround:

    expr="${test}+512)$math"
    echo "$(( expr ))"

**Comment by orbea** [comment]
Thanks, your suggestions lead to a good workaround in my script.
```
#!/bin/sh

set -eu

test='415670712'
math='21/10'

echo "$(((${test}+512)/1024*$math))" # 852448
```
 I still think the parsing error is unfortunate however.
